### ISSUE TITLE
Trigger after Restart Homey (KAKU, SolarPanels) 

### ISSUE THREAD

**Issue by Dijker** [athombv/homey#1051]
**What is your Homey version (Settings → System)?**
Homey 0.10.7
KAKU 2.6.6 (Tested with ASUN-650 , AMST-606)
Solar Panels 1.4.0 (Tested With SolarEdge)
(and maybe other apps....)

**What did you try to achieve?**
Trigger a Flow on a State Change. (fe: Door Closed, Screen Down, The Power Changed)

*Please include detailed steps how we can reproduce the issue*
1) Create a Flow with a Trigger Card for above examples. 
2) Set the (device) state, so close the door, Put the screen a little bit Down, Let the sun go down to drop to 0 Watt Power....
3) Check the Flows have triggered! (= Expected, =OK)
4) Restart Homey 

**What was the result?**
A couple of seconds after the restart of Homey all these flows trigger again!

**What did you expect as result?**
As the state didn't change again after a restart the flows should not trigger again after a restart of Homey!

**Any other remarks**
This behavior changed around 0.10.0 i think... Before this didn't happen....

As #1034 is closed and I still see this after creating a new device (ASUN650) and New test flows  a New Issue!
(like i wrote here: https://github.com/athombv/homey/issues/1034#issuecomment-256180709 )

**Comment by Dijker** [comment]
Re-tested above on clean development  Homey on v1.0.1 with an asun-650 and still a problem.

**Comment by jeroenvollenbrock** [comment]
Can you please let me know whether this issue persists or not?

**Comment by Dijker** [comment]
Hi @jeroenvollenbrock
Should it be fixed in one of the past updates?  

Yes, just tested on two Homey's, 
Prod 1.1.2 / KAKU 2.9.0 - Issue confirmed / not fixed.
R&D 1.1.4 / KAKU 3.0.0 - Issue confirmed / not fixed.

Created a new ASUN-650 on both and created a flow with the following trigger and a Push MSG
![image](https://cloud.githubusercontent.com/assets/12412734/22666362/894a0986-ecb8-11e6-81e1-3acfd93cab14.png)

Set the ASUN-650 to Up and tested the flow. OK
Restart Homey, go to Settings / Apps wait approx 30 seconds after the App started I get another Push MSG ...

**Comment by Dijker** [comment]
Hi @jeroenvollenbrock
Just tested on Homey,
Prod 1.2.0-rc8  / KAKU 3.0.1 - Issue confirmed / not fixed.

**Comment by WeeJeWel** [comment]
The problem with this issue is that a Flow triggers on a value change. So when Homey boots, the value changes from `null` to a number, boolean etc.

However, it's unclear when this is intended to trigger a Flow or not. Just after a reboot, this might not be the case. But 5, 10, or 15 minutes later? Would it still count as 'initial' value?

Saving the state of the device during a reboot is a bad idea, because then Homey has a different idea about the device's state than the app does, which can result in all sorts of problems.

Any ideas are welcome :)

**Comment by Dijker** [comment]
Yeah, I see your problem creating a solid solution. 
Although I also see a couple of questions on the forum and the above issue where it is annoying issue. 
I don't care the Solar panels where it is an extra Dot on the line and it was a pushmsg at my phone after at restart. 
But if I remember it also happens for the BitFlip and I used a virtual KAKU sometimes to have a switch to toggle where a flow switches fe all lights. 
[.... Wait....]
Just tested again with BetterLogic (rc10 - App 0.9.10 ) this doesn't happen to BitFlips (not for sure it was a problem before)
For the KAKU I Guess the App also saves the state, as if I create a virtual AC-1000 the state before the restart is visible after the restart. And Homey can't ask it the socket.....

I think the BetterLogic solution (?) I guess by not triggering on reload values after a restart (changes from $null to Value ) should be the solution. 

Maybe you could pust this @ from your shoulder to @BasKiers  and make it an App-Issue ;-)

**Comment by WeeJeWel** [comment]
I've thought of a solution and will implement it shortly :-)

**Comment by Dijker** [comment]
Thanks! I am curious how you think it can be solved. But I think many Homey users will be happy if Homey at restarts won't switch device's​ again. Pls update when I can test something at the dp channel or by siteloading an app.

**Comment by WeeJeWel** [comment]
The idea is to store the state of the device just before reboot for 30 minutes, and after reboot, every first get or realtime with the exact same value will not trigger a Flow.

**Comment by Dijker** [comment]
Great! This will fix it for planned restarts and upgrades! 
I Guess the issue will still happen  for PTP then? 
I think I can live with that,  maybe we describe that as known behavior....

**Comment by WeeJeWel** [comment]
Fixed! A device's state is synced every 10s (so even PTP's will work), and is cached for 30 minutes. Upon boot the state will be put back, and no Flows or Insights will trigger.

**Comment by Dijker** [comment]
@WeeJeWel is this available in rc6 ? Before I start testing something that wasn't included.....

**Comment by WeeJeWel** [comment]
Yep!

**Comment by Dijker** [comment]
Thanks!! Great 
(from this update on my production I can safe restart or PTP for that Homey at night when the GF is sleeping !!!)

Tested a couple of scenarios on my development 1.3.0-rc6 and it works. 
Didn't test a PTP but I trust your coding skills ;-) 

I will close for now and look for new things to keep your  work-queue  filled ;-)

**Comment by mauritsivs** [comment]
Looks like this bug is still alive. LAst night after homey updated to 1.5 it triggered a flow that turns on my TV. this flow should only trigger when an eurodomest wireless switch is switched on. happened again after rebooting homey this afternoon.

**Comment by Dijker** [comment]
Hi, 
I tried to reproduce but couldn't trigger the bug whatever I tested  
Added 3 virtual devices:
- Eurodomest 
- HE333EU
- ADC-1000

Created the flows to Pulse a LEDRing, send a Push on my phone and Log something 
none of them triggered on soft/warm restart of Homey. Not on Swithed On nor on Swithed Off.
Even removing all other Apps from the Flow to prevent a broken flow at trigger didn't work. 

Then I something unique to a otherwise empty Flow for triggering on restart to identify and test 
fe Add na action Set a Textual variable "NewVar1051"  to [[Time]]

Still didn't trigger. 
The Var should be set after restart.
Then I read back this Issue and added the ASUN-650, 
After adding that that Flow triggered and also the ADC-1000 Testflow triggerd on restart 
But looks like it doesn't always happen..... Not every restart and/or not every device.

Homey Firmware Version: 1.5.1 KAKU v3.1.5
Logs: 288FF00C6F

**Comment by bvdbos** [comment]
Same experience for me....

**Comment by elfrigo** [comment]
I've something like this issue too. I've got 9 Fibaro Smokesensors that send an alarm on a temperature of more then 55 degrees celcius. This alarm is trigger after every reboot Homey.
Quite annoying to get alarm messages in the middle of the night (when Homey gets an update).
This temperature alarm also goes off when adding the Fibaro (for the first time) to Homey. So it might be connected to the "re-adding" procedure of all z-wave devices after a reboot?
Any idea's?

**Comment by Dijker** [comment]
an other annoying one, imho the same issue:
https://forum.athom.com/discussion/comment/58524/#Comment_58524

**Comment by bvdbos** [comment]
I have one kaku-device (light) for which this happens but only after a Homey-update. Haven't been able to trigger this with a reboot or PTP. This light is connected to a virtual light button however. So might also be the virtual light. I'll delete that virtual button for the next update...

**Comment by WeeJeWel** [comment]
@bvdbos What happens when you power-off Homey for 30 minutes and then boot again?

**Comment by bvdbos** [comment]
Didn't turn on the lights. Did delete the virtual device now, I'll see what happens with the next FW-update...

**Comment by bvdbos** [comment]
Works for me now...

**Comment by Dijker** [comment]
@WeeJeWel  This one still isn't fixed imho. 
Homey 1.5.5

Added HomeEasy HE333EU
Added KAKU ASUN-650
Create Flow If HE333EU Turned On => Push MSG
Create Flow If ASUN-650 The State Changed Down => Push MSG

Set HE333EU  On and Set ASUN-650 Down, Verify you get 2 Push MSG!
Restart Homey from Settings- System, 
Verify You don't get a Push MSG, 

- You might think this one is fixed....

Get some sleep, (I guess more than 30 minutes is enough? )
Re-test: Restart Homey from Settings- System, 

>> Again 2 Push MSG  :-(  

- I guess State UP / Turned Off does the same but not within the 30 minutes after a state change? 
- As I see it with two Athom Apps KAKU and HomeEasy this could also happen for other apps/devices and maybe Virtual Devices?

- I also added a Log card (Papertrails) to these flows, they do function when testing the Flow and when triggering due to switching the device. I didn't see it in the log this was triggered when I restarted Homey and got a Push MSG so I guess flows can execute partial when not all Apps are loaded after restarting Homey.  (Could create an other Issue for that)

**Comment by WeeJeWel** [comment]
This should be fixed in 1.5.6. Any updated installed _after_ 1.5.6 should not trigger Flows by devices unnecessary.

**Comment by Dijker** [comment]
@WeeJeWel  
updated yesterday to RC1, tested several times restarting after that and it didn't trigger. 
Looked promising. 
today I updated to RC2 and the flows triggered again. (around 18u00)
after that I just updated to RC3 and the flows triggered again.

At least during updates it isn't fixed! will test restarting later...

**Comment by Dijker** [comment]
@WeeJeWel  after the Updated  to 1.5.6 rc4. Again two triggers fired.
Yhis issue is Not yet solved

**Comment by WeeJeWel** [comment]
OK, thanks for reporting!

**Comment by WeeJeWel** [comment]
Hopefully this is solved in the next release. If not, it's most likely an app-specific issue and I'll have to investigate further.

**Comment by Dijker** [comment]
Thanks a lot! 
I guess it is really fixed now. 
Last Firmware Update before was the last time I could reproduce, but since then en with the RC.5 today my Canary flows didn't trigger anymore.
